copy/copy.go

```diff
@@ -301,11 +306,33 @@ type diffIDResult struct {
 	err    error
 }
 
+func getBlob(src types.ImageSource, srcInfo types.BlobInfo) (io.ReadCloser, int64, error) {
+	if len(srcInfo.URLs) == 0 {
+		return src.GetBlob(srcInfo.Digest)
+	}
+	var (
+		resp *http.Response
+		err  error
+	)
+	for _, url := range srcInfo.URLs {
+		resp, err = http.Get(url)
```

Comment:
Implementing a specific transport like this in the generic copy code is weird. Why wouldn’t this be done by an `ImageSource`, perhaps through `ImageSource.GetBlob(info types.BlobInfo)`?